### PR TITLE
None blocking telemetry

### PR DESCRIPTION
**Pull request by george-dorin**
Add config digest to exported telemtry fields
Make telemetry collection non-blocking
https://smartcontract-it.atlassian.net/browse/BCF-2037
https://smartcontract-it.atlassian.net/browse/BCF-2127

### CODE REVIEW

File: core/services/synchronization/telem/telem.proto [hunk 1]
```diff
@@ -41,6 +41,7 @@ message TelemEnhancedEA {
     string feed=11;
     string chain_id=12;
     int64 observation=13;
-    int64 round=14;
-    int64 epoch=15;
+    string config_digest = 14;
+    int64 round=15;
+    int64 epoch=16;
```
Comment (jmank88) [inline core/services/synchronization/telem/telem.proto:46]: Is the last version already in the wild? Normally proto fields are append-only:
```suggestion
    int64 round=14;
    int64 epoch=15;
    string config_digest = 16;
```

File: core/services/ocrcommon/telemetry_test.go [hunk 2]
```diff
@@ -204,5 +243,122 @@ func TestSendEATelemetry(t *testing.T) {
...
+		},
+		lggr:               nil,
+		monitoringEndpoint: monitoringEndpoint,
+	}
+	finalResult := pipeline.FinalResult{
+		Values:      []interface{}{"123456"},
+		AllErrors:   nil,
+		FatalErrors: []error{nil},
+	}
+	wg.Add(b.N)
```
Comment (krehermann) [inline core/services/ocrcommon/telemetry_test.go:357]: i believe N is determined dynamically while running, based on how many loops can be executed in a given time. this is what i saw while writing the json benchmarks, but a i can't find a definitive reference ATM.

would be better to do wg.Add(1) inside the loop

**Review by krehermann — CHANGES_REQUESTED** [review]

File: core/services/synchronization/telem/telem.proto (hunk 1 above)
Comment (george-dorin) [reply core/services/synchronization/telem/telem.proto:46]: It's not in the wild yet, shifting the fields will not break any compatibility

File: core/services/ocrcommon/telemetry_test.go (hunk 2 above)
Comment (george-dorin) [reply core/services/ocrcommon/telemetry_test.go]: You are right, after some research I found this in the documentation:

> During benchmark execution, b.N is adjusted until the benchmark function lasts long enough to be timed reliably.

Changed to `wg.Add(1)` in [`3214452` (#8684)](https://github.com/smartcontractkit/chainlink/pull/8684/commits/3214452890eb540bf560dabd5f210b65a68697af)

File: core/services/ocrcommon/telemetry.go [hunk 3]
```diff
@@ -88,28 +88,46 @@ func getJsonParsedValue(trr pipeline.TaskRunResult, trrs *pipeline.TaskRunResult
...
+		return 0
+	}
+	return parsedValue.Int64()
+}
+
 // collectEATelemetry checks if EA telemetry should be collected, gathers the information and sends it for ingestion
 func collectEATelemetry(ds *inMemoryDataSource, trrs *pipeline.TaskRunResults, finalResult *pipeline.FinalResult) {
-	if !shouldCollectTelemetry(&ds.jb) {
+	if !shouldCollectTelemetry(&ds.jb) || ds.monitoringEndpoint == nil {
 		return
```
[the remark this answers is not in the document]
Comment (george-dorin) [reply core/services/ocrcommon/telemetry.go:120]: Added logging in [`3214452` (#8684)](https://github.com/smartcontractkit/chainlink/pull/8684/commits/3214452890eb540bf560dabd5f210b65a68697af)

**Review by krehermann — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**